src/unity/Assets/Crest/Scripts/Collision/CollProviderDispTexs.cs

```diff
@@ -13,25 +13,31 @@ public class CollProviderDispTexs : ICollProvider
     {
         int _areaLod = -1;
 
-        public bool SampleDisplacement(ref Vector3 worldPos, ref Vector3 displacement)
+        public bool SampleDisplacement(ref Vector3 in__worldPos, out Vector3 displacement)
         {
-            int lod = LodDataAnimatedWaves.SuggestDataLOD(new Rect(worldPos.x, worldPos.z, 0f, 0f), 0f);
-            if (lod == -1) return false;
-            return OceanRenderer.Instance._lodDataAnimWaves[lod].SampleDisplacement(ref worldPos, ref displacement);
+            int lod = LodDataAnimatedWaves.SuggestDataLOD(new Rect(in__worldPos.x, in__worldPos.z, 0f, 0f), 0f);
+            if (lod == -1) {
+                displacement = Vector2.zero;
+                return false;
+            }
+            return OceanRenderer.Instance._lodDataAnimWaves[lod].SampleDisplacement(ref in__worldPos, out displacement);
         }
-        public bool SampleDisplacement(ref Vector3 worldPos, ref Vector3 displacement, float minSpatialLength)
+        public bool SampleDisplacement(ref Vector3 in__worldPos, out Vector3 displacement, float minSpatialLength)
         {
             // select lod. this now has a 1 texel buffer, so the finite differences below should all be valid.
-            PrewarmForSamplingArea(new Rect(worldPos.x, worldPos.z, 0f, 0f), minSpatialLength);
+            PrewarmForSamplingArea(new Rect(in__worldPos.x, in__worldPos.z, 0f, 0f), minSpatialLength);
 
-            return SampleDisplacementInArea(ref worldPos, ref displacement);
+            return SampleDisplacementInArea(ref in__worldPos, out displacement);
         }
 
-        public bool SampleHeight(ref Vector3 worldPos, ref float height)
+        public bool SampleHeight(ref Vector3 in__worldPos, out float height)
         {
-            int lod = LodDataAnimatedWaves.SuggestDataLOD(new Rect(worldPos.x, worldPos.z, 0f, 0f), 0f);
-            if (lod == -1) return false;
-            height = OceanRenderer.Instance._lodDataAnimWaves[lod].GetHeight(ref worldPos);
+            int lod = LodDataAnimatedWaves.SuggestDataLOD(new Rect(in__worldPos.x, in__worldPos.z, 0f, 0f), 0f);
+            if (lod == -1) {
+                height = float.NaN;
```

Comment:
NaN's terrify me - maybe an irrational fear but in previous projects I've seen them spread and cause badness around a system, once we had crashes on some PCs due to NaNs infecting the graphics driver. I'd probably feel more comfortable going with 0 here, or do you feel strongly it should be NaN? Zero would also match the Vector3.zero.

Reply:
Yeah, I'm happy changing that.

I would be nice if there was a system where you could turn on NaN's in debug builds only, but also with a feature that tagged every NaN so you could see where they spread from.

But having less potential badness in a system makes sense.